### DISCUSSION TITLE
productId auto increment to random

### DISCUSSION THREAD

**Discussion by KushagraVasal** [vendurehq/vendure#2150] [Q&A]
How can i convert auto_increment productId into nanoid productId

**Comment by KushagraVasal** [comment]
When I tried to use entityOptions
I encountered an error "SqliteError: datatype mismatch"
```
export const config: VendureConfig = {
    entityOptions: {
    	entityIdStrategy: new UuidIdStrategy(),
    },
```

**Reply by michaelbromley** [reply]
Hi,
if you change the entity id strategy, then the existing entities in your DB will have integer IDs, but now the DB expects string IDs - this is likely causing your error. You'll need to re-created the database.

**Reply by KushagraVasal** [reply]
I am using "npx @vendure/create@next my-shop" how can i re-created database?

**Reply by michaelbromley** [reply]
See this answer: https://github.com/vendure-ecommerce/vendure/discussions/2152#discussioncomment-5751254

**Comment by KushagraVasal** [comment]
uuid worked but i am planning to use nanoid can you help me with that
